src/dict_list.jl

```diff
@@ -275,3 +275,82 @@ julia> dict_list(d) # only in case `:a` is `1` the dictionary will get key `:c`
      end
      :(toDependentParameter($(esc(value)),$(esc(condition))))
  end
+
+"""
+   ComputedParameter{T}
+Type that holds the name of a parameter (independentP) and a function (func). After the possible parameter combinations are created, dict_list will replace instances of ComputedParameter to the result of the function func, evaluated with the value of the parameter independentP.
+
+Examples:
+```
+julia> p = Dict(:α => [1, 2],
+           :solver => ["SolverA","SolverB"],
+           :β => ComputedParameter(:α, x -> x^2),
+           )
+Dict{Symbol, Any} with 3 entries:
+  :α      => [1, 2]
+  :solver => ["SolverA", "SolverB"]
+  :β      => ComputedParameter{Symbol}(:α, #51)
+
+julia> dict_list(p)
+4-element Vector{Dict{Symbol, Any}}:
+ Dict(:α => 1, :solver => "SolverA", :β => 1)
+ Dict(:α => 2, :solver => "SolverA", :β => 4)
+ Dict(:α => 1, :solver => "SolverB", :β => 1)
+ Dict(:α => 2, :solver => "SolverB", :β => 4)
+```
+
+A vector of parameter names can also be passed when the accompanying function uses multiple arguments:
+```julia
+ julia> p2 = Dict(:α => [1, 2],
+           :β => [10,100],
+           :solver => [SolverA,SolverB],
+           :γ => ComputedParameter([:α,:β], (x,y) -> x^2 + 2y),
+           )
+Dict{Symbol, Any} with 4 entries:
+  :α      => [1, 2]
+  :γ      => ComputedParameter{Symbol}([:α, :β], #7)
+  :solver => DataType[SolverA, SolverB]
+  :β      => [10, 100]
+
+julia> dict_list(p2)
+8-element Vector{Dict{Symbol, Any}}:
+ Dict(:α => 1, :γ => 21, :solver => SolverA, :β => 10)
+ Dict(:α => 2, :γ => 24, :solver => SolverA, :β => 10)
+ Dict(:α => 1, :γ => 21, :solver => SolverB, :β => 10)
+ Dict(:α => 2, :γ => 24, :solver => SolverB, :β => 10)
+ Dict(:α => 1, :γ => 201, :solver => SolverA, :β => 100)
+ Dict(:α => 2, :γ => 204, :solver => SolverA, :β => 100)
+ Dict(:α => 1, :γ => 201, :solver => SolverB, :β => 100)
+ Dict(:α => 2, :γ => 204, :solver => SolverB, :β => 100)
+```
+"""
+struct ComputedParameter{T}
+    independentParam::Vector{T}
+    func::Function
+end
+
+"""
+    ComputedParameter(independentP <: Union{String,Symbol}, func :: Function)
+Constructs a ComputedParameter from a single independent parameter.
+"""
+function ComputedParameter(independentP :: Union{String,Symbol}, func :: Function) 
+    return ComputedParameter([independentP], func)
+end
+
+export ComputedParameter
```

Comment:
Also please move this to the top of the file with the other export statements

Reply:
Oh yes. Actually, this file did not have the exports at the top. I took the liberty of moving them up.